### PR TITLE
ci: gate PRs on a `core` pytest marker plus changed tests; full suite in test-full

### PR DESCRIPTION
**Pull request by tgvashworth**
## Summary

CI test time on a pull request was about 21 minutes, all of it blocking. This splits the Tests workflow into a required core job and a non-blocking full run.

- **`test` (required) runs `pytest -m core` plus the tests the PR touched.** `core` is a registered pytest marker (with `--strict-markers`, so a typo fails collection). 886 tests carry it: the safety checks (CSRF, admin login, the state machine, the task queue), the query-count snapshots, and one set of end-to-end flows per edition. Locally the core run finishes in under three minutes on four workers. The job name is unchanged, so the required-check ruleset needs no edit.
- **The PR's own tests always run in the required job.** `scripts/changed_test_paths.py` lists the test files a PR added or modified, the directory of any changed `conftest.py`, and the trees whose conftest imports a changed fixtures module (so editing `n26/tests/fixtures.py` runs all three n26 trees; a deleted conftest still counts, and production modules the root conftest imports do not). The root conftest reads that list from `GYRINX_CHANGED_TEST_PATHS` and marks those tests `core` during collection. The rule is a pure function with its own tests.
- **`test-full` runs everything but does not block.** Same setup as before, minus Bandit. A new push to the PR cancels the run in flight; on main every push gets its own run.
- **A bounds check keeps the core suite honest.** The `test` job fails if `-m core` collects fewer than 300 or more than 1200 tests, so the marker set cannot drift to empty or grow back into the whole suite unnoticed.
- **Shared setup moved to a composite action** at `.github/actions/python-env` (Python, uv, `.env`, the Postgres lock tuning), used by all three jobs in `test.yaml` and by `ci-checks.yml`. Bandit moved to ci-checks so it no longer queues behind the tests.

Posting to Slack when `test-full` fails on main is left as a TODO in the workflow.

Builds on #2420, which set `-n logical` on the old job; both new jobs carry the same flag.

## Test plan

- [ ] On this PR, the `test` job's "Check the core suite" step reports a count in the 300..1200 range
- [ ] The "List the tests this pull request touched" step lists the marked files and `gyrinx/tests/test_changed_test_paths.py`
- [ ] `test` finishes in a few minutes; `test-full` and `fresh-database` pass
- [ ] `ci-checks` runs Bandit and passes
- [ ] After merge, the ruleset's required checks (`test`, `ci-checks`, `fresh-database`) still resolve

## How to try it locally

```bash
pytest -m core                          # what the required job runs
scripts/changed_test_paths.py origin/main   # what a branch would pull in
GYRINX_CHANGED_TEST_PATHS="n23/core/tests/test_clone.py" pytest -m core --collect-only -q | grep -c test_clone
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_018GsdfBg1yW7PoKwVJetAup

### CODE REVIEW

File: scripts/changed_test_paths.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+def changed_files_against(base: str) -> list[str]:
+    out = subprocess.run(  # nosec B607 — fixed argv; git resolved from PATH like every repo tool
+        ["git", "diff", "--name-only", "--diff-filter=AMR", f"{base}...HEAD"],
+        check=True,
+        capture_output=True,
+        text=True,
+        cwd=ROOT,
+    ).stdout
+    return [line for line in out.splitlines() if line and (ROOT / line).exists()]
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply scripts/changed_test_paths.py]: Fixed in 2c0dd52f9: deletions are diffed too; a deleted conftest or fixtures module still lists its tree, a deleted test file is dropped.

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -34,58 +39,101 @@ jobs:
...
-      - name: Write to .env file
+      - name: Check the core suite is neither empty nor the whole suite
+        shell: bash
         run: |
-          echo "SECRET_KEY=test-secret" >> .env
-          echo "DJANGO_SUPERUSER_PASSWORD=password" >> .env
+          source .venv/bin/activate
+          # A collection error fails the step here; a clean listing with no
+          # matches must still reach the bounds message, hence `|| true`.
+          listing=$(pytest -m core --collect-only -q -n 0)
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply .github/workflows/test.yaml]: Fixed in 006ea93dc: exit 5 is tolerated and reaches the bounds message; any other status fails the step.

File: scripts/changed_test_paths.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+def changed_files_against(base: str) -> list[str]:
+    """Every path the change added, modified, renamed or deleted."""
+    out = subprocess.run(  # nosec B607 — fixed argv; git resolved from PATH like every repo tool
+        ["git", "diff", "--name-only", "--diff-filter=AMRD", f"{base}...HEAD"],
+        check=True,
+        capture_output=True,
+        text=True,
+        cwd=ROOT,
+    ).stdout
+    return [line for line in out.splitlines() if line]
```
[the remark this answers is not in the document]
Comment (tgvashworth) [reply scripts/changed_test_paths.py:158]: Fixed in 006ea93dc: the diff runs with --no-renames, so a move is a delete plus an add and the old subtree is listed.
